eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -1343,27 +1343,46 @@ passTick = do
   impLastTickL += 1
   impNESL .= nes
 
+-- | Win with supplied probability
+gambleWithProbability ::
```

Comment:
Perhaps `trueWithProbability` would be clearer? Or `drawWithProbability` / `sampleWithProbability`? 
I don't mind `gambleWithProbability` either, just an idea.

This seems like a useful function to have!

Reply:
Yeah, I struggled with a name for it. Thanks to your suggestion I think `drawBoolWithProbability` will be a good name for it.